lib/reek/spec/should_reek_only_of.rb

```diff
@@ -15,25 +15,30 @@ def matches?(actual)
       def matches_examiner?(examiner)
         @examiner = examiner
         @warnings = @examiner.smells
-        @warnings.length == 1 && @warnings[0].matches?(@klass, @patterns)
+        return false unless @warnings.length == 1
+        @warning = @warnings[0]
+        @warning.matches?(@smell_category, @smell_details)
```

Comment:
I think these three lines should be replaced by `@warnings.all? { |w| w.matches?(@smell_category) }`.